File: pathfinder-api/src/main/java/de/cubbossa/pathfinder/group/FindDistanceModifier.java
package de.cubbossa.pathfinder.group;

import de.cubbossa.pathfinder.misc.NamespacedKey;

public interface FindDistanceModifier extends Modifier {

  NamespacedKey KEY = NamespacedKey.fromString("pathfinder:find-distance");

  @Override
  default NamespacedKey getKey() {
    return KEY;
  }

  double distance();
}

File: pathfinder-api/src/main/java/de/cubbossa/pathfinder/group/FindDistanceModifier.kt
package de.cubbossa.pathfinder.group

import de.cubbossa.pathfinder.misc.NamespacedKey
import de.cubbossa.pathfinder.misc.NamespacedKey.Companion.fromString

interface FindDistanceModifier : Modifier {
    val distance: Double

    companion object {
        val key: NamespacedKey = fromString("pathfinder:find-distance")
    }
}
